proposals/csharp-12.0/collection-expressions.md

```diff
@@ -314,6 +314,54 @@ If the target type is an *array*, a *span*, a type with a *[create method](#crea
     List<int> z = [];
     ```
 
+## Spreads
+[spreads]: #spreads
+
+```csharp
+bool b = true;
+int[] x = [.. []];            // x = []
+byte[] y = [.. b ? [1] : []]; // y = [(byte)1]
+```
+
+A *spread element* `..E` is applicable as an iterator of type `T` if any of the following hold:
+
+* There is an implicit conversion from the [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement) of `E` to `T`.
+* `E` is a *collection expression* and for each element `Eᵢ` there is an implicit conversion to `T`.
+* `E` is a [*conditional expression*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#1115-conditional-operator) `b ? x : y` where one or both of `x` and `y` is a *collection expression* and for each of the operands `x` and `y` one of the following hold:
+  * The operand is a collection expression and for each element `Eᵢ` there is an implicit conversion to `T`.
+  * The operand has an [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement) `Tₑ` and there is an implicit conversion from `Tₑ` to `T`.
+
+For a collection expression used a spread element expression, or for a collection expression as the second or third operand in a conditional expression used as a spread element expression, the compiler may use any conforming representation for the collection instance, including eliding the collection.
+
+For example, the code generated for `List<int> list = [x, y, .. b ? [z] : []]` could avoid an intermediate collection:
+```c#
+List<int> list = new();
+list.Add(x);
+list.Add(y);
+if (b) list.Add(z);
+```
+
+*Should `switch` expressions be supported in spread elements?*
```

Comment:
A strong correspondence in my mind is that `a ? b : c` is equivalent to `s switch { true => b, false => c, }`.  Every time that is not true makes me sad :)

Reply:
I agree. I think switch expressions in many ways supercede the conditional operator, so I've practically stopped using the conditional operator unless a) it's comparing non-constants, or b) it's so short that using it inline is reasonable.